kong/db/strategies/postgres/connector.lua

```diff
@@ -314,55 +322,94 @@ function _mt:init()
 end
 
 
-function _mt:init_worker(strategies)
-  if ngx.worker.id() == 0 then
+local function cleanup_expired_rows_in_table(config, table_name)
+  -- Create new connection on each table to avoid reusing a connection that might be already timed out
+  local connector = connect(config)
+
+  local cleanup_start_time = now_updated()
+  local ttl_escaped = connector:escape_identifier("ttl")
+  local expired_at_escaped = connector:escape_identifier("expire_at")
+  local column_name = table_name == "cluster_events" and expired_at_escaped
+                                                      or ttl_escaped
+
+  local cleanup_statement = concat {
+    "DELETE FROM ",
+    connector:escape_identifier(table_name),
+    " WHERE ctid in (SELECT ctid FROM ",
+    connector:escape_identifier(table_name),
+    " WHERE ",
+    column_name,
+    " < TO_TIMESTAMP(%s) AT TIME ZONE 'UTC' LIMIT ",
+    EXPIRED_ROW_BATCH_SIZE,
+    ")",
+  }
 
-    local table_names = get_names_of_tables_with_ttl(strategies)
-    local ttl_escaped = self:escape_identifier("ttl")
-    local expire_at_escaped = self:escape_identifier("expire_at")
-    local cleanup_statements = {}
-    local cleanup_statements_count = #table_names
-    for i = 1, cleanup_statements_count do
-      local table_name = table_names[i]
-      local column_name = table_name == "cluster_events" and expire_at_escaped
-                                                          or ttl_escaped
-      cleanup_statements[i] = concat {
-        "  DELETE FROM ",
-        self:escape_identifier(table_name),
-        " WHERE ",
-        column_name,
-        " < CURRENT_TIMESTAMP AT TIME ZONE 'UTC';"
-      }
+  cleanup_statement = fmt(cleanup_statement, connector:escape_literal(tonumber(fmt("%.3f", cleanup_start_time))))
+
+  local cleanup_loop = 0
+  while cleanup_loop <= EXPIRED_ROW_CLEANUP_LOOP_MAX do
+    cleanup_loop = cleanup_loop + 1
+    local ok, err = connector:query(cleanup_statement)
+    if not ok then
+      if err then
+        log(WARN, "failed to clean expired rows from table '",
+                  table_name, "' on PostgreSQL database (",
+                  err, ")")
+      else
+        log(WARN, "failed to clean expired rows from table '",
+                  table_name, "' on PostgreSQL database")
+      end
+    end
+
+    if ok.affected_rows == 0 then
```

Comment:
Maybe we could use `ok.affected_rows <= (10% of EXPIRED_ROW_BATCH_SIZE)` so that we will never hit an unnecessary loop especially an unnecessary query.
If every time there is a 10% left, it could be cleared at next window. Even we may consider using `ok.affected_rows <  EXPIRED_ROW_BATCH_SIZE` for making use of the capacity of batch fully.

Reply:
Good suggestion, I've changed to 
```
tonumber(ok.affected_rows) < EXPIRED_ROW_BATCH_SIZE
```
to check when the last loop happens